File: problems/c/cups/cups.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    n = int(input())

    size_color = dict()
    for _ in range(n):
        cr = input().split()
        try:
            c = cr[0]
            r = int(cr[1])
        except:
            c = cr[1]
            r = float(int(cr[0])/2)
        size_color[r] = c

    for size in sorted(size_color.keys()):
        print(size_color[size])

if __name__ == '__main__':
    main()

File: problems/c/cups/cups.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <bits/stdc++.h>

using namespace std;

int main() {
    int n;
    int radius;
    string color;
    map<int, string> tokens;

    cin >> n;
    cin.ignore();

    for (int i=0; i<n; i++) {
        for (int j=0; j<2; j++) {
            string token;
            stringstream ss;
            int val;

            cin >> token;
            ss << token;

            if (ss >> val) {
                if (j == 0) {
                    radius = val / 2;
                }
                else {
                    radius = val;
                }
            }
            else {
                color = token;
            }
        }
        tokens[radius] = color;
    }

    for (pair<int, string> x : tokens)
    {
        cout << x.second << endl;
    }
}